service/controller/key/key.go

```diff
@@ -293,8 +293,14 @@ func IsSucceededProvisioningState(s string) bool {
 	return s == "Succeeded"
 }
 
-func MachinePoolOperatorVersion(cr expcapzv1alpha3.AzureMachinePool) string {
-	return cr.GetLabels()[LabelOperatorVersion]
+// These are the same labels that kubernetesd adds when creating/updating an AzureConfig.
+func KubeletLabelsNodePool(getter LabelsGetter) string {
+	var labels string
+
+	labels = ensureLabel(labels, label.Provider, "azure")
+	labels = ensureLabel(labels, label.OperatorVersion, OperatorVersion(getter))
```

Comment:
We should slowly start moving these labels to `apiextensions/pkg/label`.

Reply:
Not a blocker.

Reply:
Yeah, let's do it in a different PR though.